src/search/airportsearch.cpp

```diff
@@ -744,14 +750,166 @@ void AirportSearch::randomFlightplanClicked()
+        float lonX, latY;
+        do
+        {
+          if(triedIndexDestination.count() == countResult)
+            goto destinationsEnd;                                                 // all destinations have been depleted, try a different departure
+          if(triedIndexDestination.count() < randomLimit)                         // the "if" is inside the "while" for the destination because the indices get used up during "picking"
+          {
+            do
+            {
+              indexDestination = QRandomGenerator::global()->bounded(countResult);
+            }
+            while(triedIndexDestination.contains(indexDestination));
+          }
+          else
+          {
+            indexDestination = QRandomGenerator::global()->bounded(countResult);
+            while(triedIndexDestination.contains(indexDestination))
+            {
+              indexDestination = (indexDestination + 1) % countResult;
+            }
+          }
+          triedIndexDestination.insert(indexDestination, true);
+          lonX = data[indexDestination].second.getLonX();
+          latY = data[indexDestination].second.getLatY();
+        }
+        while(lonX == atools::geo::Pos::INVALID_VALUE || latY == atools::geo::Pos::INVALID_VALUE);
+        const double lon2 = lonX * degToRad;
+        const double lat2 = latY * degToRad;
+        dist = R_earth * qAcos(qSin(lat1) * qSin(lat2) + qCos(lat1) * qCos(lat2) * qCos(lon2 - lon1));     // http://www.movable-type.co.uk/scripts/latlong.html Spherical Law of Cosines
+        destinationSuccess = true;
+      }
+      while(dist < distanceMin || dist > distanceMax);
+destinationsEnd:
+      if(destinationSuccess)                                                      // the last triedIndexDestination might be taken but it might have passed the last condition
+      {
+        noSuccess = false;
+        continue;
+      }
+    }
+    while(noSuccess);
+allDeparturesTried:
+    if(departureSuccess)                                                          // the last triedIndexDeparture might be taken but it might have passed the last condition, destinationSuccess must be true if this "if" is reached
+    {
+      qDebug() << Q_FUNC_INFO << "random flight, index departure: " << indexDeparture
+               << ", random flight, index destination: " << indexDestination;
+
+      map::MapAirport airportDeparture;
+      map::MapAirport airportDestination;
+
+      AirportQuery *airportQuery = NavApp::getAirportQuerySim();
+
+      airportQuery->getAirportById(airportDeparture, data[indexDeparture].first);
+      airportQuery->getAirportById(airportDestination, data[indexDestination].first);
+
+      NavApp::getMainWindow()->routeNewFromAirports(airportDeparture, airportDestination);
+    }
+    else
+    {
+      QMessageBox msgBox;
+      msgBox.setText(tr("No airports found in the search result satisfying the criteria."));
+      msgBox.exec();
+    }
+  }
+  else
+  {
+    QMessageBox msgBox;
```

Comment:
I'd not use blocking message boxes in this case. These are quite annoying and I try to avoid them wherever possible. 

Simply avoid min going over max. You can do this by connecting to `QSpinBox::valueChanged()` and then using `setMinimum()` to the max spin box when getting a signal from the min box and using`setMaximum` to the min spin box when getting a signal from the max box. Values adapt automatically when doing this.
This behavior is then consistent with the other min/max widgets.

Reply:
I implemented this. The singleShot is for preventing typing 19 in max when min is 100 and one wanted to type 190 causing a "reset". The static int is to only take the last valueChanged into account (ie dismiss earlier singleShots which would callback during the current new timeout period).

Reply:
No idea why you would need a timer. It is working without for the runway and altitude widgets.

https://github.com/albar965/littlenavmap/blob/e64061b8f9e9ac1e11952e5a4257cafa2803edf5/src/search/searchbasetable.cpp#L410-L425

https://github.com/albar965/littlenavmap/blob/e64061b8f9e9ac1e11952e5a4257cafa2803edf5/src/search/searchbasetable.cpp#L483-L513

Reply:
you have editStartTimer() which might be a different timer solution for the same issue. Other than this you use the same event as I did and you don't have the issue but i have and i don't know how to solve it without some sort of waiting for further input.

Reply:
The timer is only for a delayed update especially for the slow distance search.

I can simply merge it and have a look at it. The Qt GUI stuff can be a major pain in the rear in some cases. Half of layout management consists of workarounds, for example. :frowning_face:

Reply:
yes, please merge and have a look.